### PR TITLE
fix searchNextLyrics to only find lyrics in same voice (i.e. track)

### PR DESCRIPTION
**Pull request by stbuehler**
- fixes display bugs when hyphens/lines would continue in a different voice after line breaks

### CODE REVIEW

**Comment by lasconic** [comment]
Thank you for making several PR. In order to merge them, I need you to sign the MuseScore CLA at https://musescore.org/cla

**Comment by stbuehler** [comment]
Why would I give "Werner Schweer" my contributions under a "do anything with it" license, but he puts his work under GPL?

I'll have to think about that (most of my own projects are MIT / BSD, so I usually don't give a shit about it, but this CLA doesn't sound fair).

**Comment by lasconic** [comment]
Sure. 
Some more context about why the CLA exists http://dev-list.musescore.org/MuseScore-to-mobile-td6232012.html Also you might want to read https://musescore.org/en/about/license-policy

**Comment by stbuehler** [comment]
I see no reason to support pushing software to a platform which doesn't allow GPL.

I think GPL is basically about **preventing** this use case in the first place, and you should reconsider your choice of license if that doesn't represent what you actually wanted.

I'm not gonna sign it, although it saddens me not to be able to improve the quality of MuseScore, especially as it didn't come to my mind that any normal (i.e. not obviously commercial) project would require such CLA and I already put some effort into fixing things.

**Comment by thomasbonte** [comment]
@stbuehler Hi Stefan, thank you for sharing your concerns with the MuseScore CLA. 

Allow me to answer on your last line: most open source projects of any significance have CLA's even place including the free software foundation managing the GNU project. Here is a non-exhaustive list: https://en.wikipedia.org/wiki/Contributor_License_Agreement#Rationale

As for why MuseScore has a CLA. MuseScore exists today because there is a business model which revolves around a cloud service and mobile apps. Because of that, several people can work full time on the open source project, continuously improving the MuseScore software and infrastructure. This would not have been possible without a CLA in place. 

Basically is an agreement between the MuseScore lead developer Werner Schweer and the code contributors. As the contributors sign the CLA, they give Werner the opportunity to keep on working on MuseScore. Without that, Werner would have moved on to something else and there was no MuseScore as you know today.

Hope this answer may address your concerns.

**Comment by stbuehler** [comment]
@thomasbonte There are very different kinds of CLAs. Usually they just write down what other people would simply assume to be given. For example I consider my patches to be released under the same license as the source code I submit them to already is (in this case GPL2), that they are provided on an "AS IS" basis, that they are either my own contributions or I have the rights to submit. And that obviously the maintainers decide what gets accepted upstream (Actually point 4 of this fucked up CLA could be interpreted **very** differently. "any project or source repository" without any restriction could also mean my fork of MuseScore, but also every other project owned by anyone completely unrelated to MuseScore. I don't see why it was added in the first place, and worded so badly).

So just because many projects use CLAs does in no way justify point 1 of the MuseScore CLA. There are other projects which ask for similar agreements (I guess Qt does), but I'd probably won't invest any time in contributing to those anymore either (I think Trolltech didn't ask for a CLA back in 2007 when I sent them a patch).

In short: if you want to make money of your project, don't expect me to grant you rights on my patches you won't grant me on your part of the source.

And be careful with the "continuously improving the MuseScore software and infrastructure" argument, because the bugs my patches would fix are kinda.. well. I'd say the code is in a very sad state, and my patches were rather low-hanging fruits.

Just a hint how to get a perhaps fairer license model:

- release the rendering core as MIT/BSD (dual-license GPL until Qt dependency is removed)
- Qt desktop frontend as GPL
- mobile apps as closed source: you sell it, you code it. anyone else could do the same.

**Comment by thomasbonte** [comment]
Hi Stefan, I appreciate you challenge the status quo but I think the
current setup is already fair and balanced. So I rest my communication
here. Good luck.
